src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexNode.cs

```diff
@@ -1238,20 +1446,21 @@ public RegexNode MakeQuantifier(bool lazy, int min, int max)
 
         public void AddChild(RegexNode newChild)
         {
-            RegexNode reducedChild = newChild.Reduce();
-            reducedChild.Next = this;
+            newChild.Next = this; // so that the child can see its parent while being reduced
```

Comment:
does this mean that `.Next` could be renamed `.Parent` now?

Reply:
I don't know.  It's still on my list of things to look at cleaning up at some point.  Once the tree is fully formed, Next means Parent.  I don't know if elsewhere in the parser Next ever means something different.